references/validationFunctions/treatment/checkWhenNoTreatment.js

```diff
@@ -5,42 +5,81 @@
+ * TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+ * OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+ * IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
  * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
- *  
+ *
  *
  */
 
 /**
- * If treatment_type is 'No treatment', core treatment fields should not be submitted.
+ * If treatment_type is 'No treatment' or 'Unknown', core treatment fields should not be submitted.
  */
 
-const validation = () => 
-  (function validate(inputs) {
-      const {$row, $name, $field} = inputs;
-      let result = {valid: true, message: "Ok"};
-      const coreFields = ['treatment_start_interval', 'treatment_duration', 'is_primary_treatment', 'treatment_intent', 'treatment_setting', 'response_to_treatment_criteria_method', 'response_to_treatment'];
- 
-      // checks for a string just consisting of whitespace
-      const checkforEmpty = (entry) => {return /^\s+$/g.test(decodeURI(entry).replace(/^"(.*)"$/, '$1'))};
-      const treatmentType = ($row.treatment_type).map(value => value.toLowerCase());
-       
-      if (!treatmentType.includes("no treatment") && coreFields.includes($name) && (!$field || $field === null || checkforEmpty($field))) {
-          result = { valid: false, message: `The '${$name}' field must be submitted when the 'treatment_type' field is '${treatmentType}'`};
-      }
-      else if (treatmentType.includes("no treatment") && ($field && $field != null && !(checkforEmpty($field)))) {
-        if (coreFields.includes($name) || (typeof($field) === 'string' && $field.trim().toLowerCase() != 'not applicable') || typeof($field) === 'number') {
-          result = { valid: false, message: `The '${$name}' field cannot be submitted if the 'treatment_type' field is '${treatmentType}'`};
+const validation = () =>
+    (function validate(inputs) {
+        const { $row, $name, $field } = inputs;
+        const result = { valid: true, message: 'Ok' };
+
+        const arrayItemsInSecondArray = (arr1, arr2) => {
+            return arr2.some(arr2Item => {
+                return arr1.includes(arr2Item);
+            });
+        };
+
+        const coreFields = [
+            'treatment_start_interval',
+            'treatment_duration',
+            'is_primary_treatment',
+            'treatment_intent',
+            'treatment_setting',
+            'response_to_treatment_criteria_method',
+            'response_to_treatment',
+        ];
+
+        const treatmentExceptionTypes = ['no treatment', 'unknown'];
+
+        // checks for a string just consisting of whitespace
+        const checkforEmpty = entry => {
+            return /^\s+$/g.test(decodeURI(entry).replace(/^"(.*)"$/, '$1'));
+        };
+        const treatmentTypes = $row.treatment_type.map(value => value.toLowerCase());
+
+        const recordHasTreatments = !arrayItemsInSecondArray(
```

Comment:
this logic checks the records' treatments, and identifies those that don't have any (upon the logic that "no treatment" and "unknown" are the absence of treatment)